vms/platformvm/block/builder/builder.go

```diff
@@ -342,3 +401,17 @@ func getNextStakerToReward(
 	}
 	return ids.Empty, false, nil
 }
+
+type stateGetter struct {
+	state state.Chain
+}
+
+func (s stateGetter) GetState(ids.ID) (state.Chain, bool) {
+	return s.state, true
+}
+
+func wrapState(parentState state.Chain) (state.Diff, error) {
+	return state.NewDiff(ids.Empty, stateGetter{
+		state: parentState,
+	})
+}
```

Comment:
I think we've implemented this elsewhere. Should we enshrine this ability somewhere? Perhaps calling it something like `state.NewDiffOn` or something

Reply:
Like that